### PR TITLE
Install crossgen, jitinterface, clrjit into the sharedframework folder in crosscomponent build.

### LINKED ISSUE #50979: x64-hosted crossgen.exe missing in Microsoft.NETCore.App.Runtime.win-arm64 package
Might be caused by #49906. @jkoritzinsky
```
D:\workspace\_work\1\s\src\Framework\App.Runtime\src\Microsoft.AspNetCore.App.Runtime.csproj(399,5): error : Could not find crossgen D:\workspace\_work\1\s\.packages\microsoft.netcore.app.runtime.win-arm64\6.0.0-preview.4.21208.6\tools\x64_arm64\crossgen.exe
```

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Fixes https://github.com/dotnet/runtime/issues/50979

### CODE REVIEW

File: src/coreclr/components.cmake [hunk 1]
```diff
@@ -19,3 +19,4 @@ add_dependencies(runtime jit iltools)
 
 # The cross-components build is separate, so we don't need to add a dependency on coreclr_misc
 add_component(crosscomponents)
+add_dependencies(crosscomponents jit)
```
Comment (hoyosjs) [inline src/coreclr/components.cmake]: Adding this dependency breaks the crossdac build, which is considered a crosscomponent. It also tries to run shell scripts on linux when running `./src/coreclr/build-runtime.cmd checked x64 -linuxdac`. We'd need to exclude building the JIT under the crossdac build.

**Review by hoyosjs — COMMENTED** [review]
The only thing I don't see in this list is the cross bitness dac + dbi and the long name cross bit dac, which also used to exist in the arch/sharedFramework directory for windows. Can you please add these too?

**Review by AntonLapounov — APPROVED** [review]

**Comment by AntonLapounov** [comment]
Thank you!

**Comment by sebastienros** [comment]
In case it's not a very straightforward fix (I see it's failing the build), can we revert the original PR in the meantime? It's now also blocking sdk releases from ingesting aspnet.

**Comment by jkoritzinsky** [comment]
@AntonLapounov can you look at the most recent commit? I don't think jitinterface_x86/64 needs to be in the folder since that's only used by crossgen2, which doesn't ship in the Runtime package (only crossgen1 does).

**Comment by AntonLapounov** [comment]
That is correct.  I mentioned the issue with the jitinterface library in https://github.com/dotnet/runtime/issues/50979#issuecomment-816352434.

**Review by hoyosjs — APPROVED** [review]

**Pull request merged**